api/plugins/ginfo/ginfo.py

```diff
@@ -0,0 +1,153 @@
+
+"""
+  Integration into the Ginfo Map via Firebase REST API
+"""
+from django.http import HttpResponse, HttpRequest
+from django.db.models import Q
+import requests
+import math
+from api.models import Character
+from ginfocharacter import GinfoCharacter
+from datetime import datetime
+from firebase_pushid import PushID
+
+
+class GinfoPlugin(object):
+    """
+      GinfoPlugin provides methods to post a position to the ginfo firebase database
+    """
+    API_URL = 'https://ginfo-34baa.firebaseio.com'
+    GAME = 'conanexiles'
+    MAP = 'conanexiles_desert'
+    GINFO_USER_UID = 'y67Efa3kddSuhb26UoUapIJakMh1'
+
+    def __init__(self):
+        self.uid_generator = PushID()
+
+    def marker_path(self, group, marker):
+        """
+            Returns the path to which a marker has to be posted in the firebase database
+
+            @param group: UID of the ginfo group for which the marker should be posted
+            @param marker: UID of the marker that should be posted
+        """
+        return "games/" + self.GAME + "/groups/" + group + "/maps/" + self.MAP + '/markers/' + marker
```

Comment:
Since strings are immutable in python, using format is a lot faster. This allocates a new string for every + operation.

```python
 return 'games/%s/groups/%s/maps/%s/markers/%s' % (self.GAME, group, self.MAP, marker)
```